modules/api/php/endpoints/candidate/visit/visit.class.inc

```diff
@@ -147,16 +150,18 @@ class Visit extends Endpoint implements \LORIS\Middleware\ETagCalculator
     /**
      * Generates a response fitting the API specification for this endpoint.
      *
+     * @param string $version The version of the API
+     *
      * @return ResponseInterface The outgoing PSR7 response
      */
-    private function _handleGET(): ResponseInterface
+    private function _handleGET(string $version): ResponseInterface
     {
         if ($this->_visit === null) {
             return new \LORIS\Http\Response\JSON\NotFound('Visit not found');
         }
 
         return new \LORIS\Http\Response\JsonResponse(
-            (new \LORIS\api\Views\Visit($this->_visit))
+            (new \LORIS\api\Views\Visit($this->_visit, $version))
```

Comment:
This seems to be a good use case for polymorphism. 
I think `\LORIS\api\Views\Visit_0_0_2` would do the trick.

Reply:
I'm not sure of what you are suggesting, you mean we could create a new Class `\LORIS\api\Views\Visit` that would take `$version` as a parameter? If it is polymorphism, shouldn't it also be called `\LORIS\api\Views\Visit` instead of `\LORIS\api\Views\Visit_0_0_2`? Plus I think this solution (with the modifications of `modules/api/php/views/visit.class.inc` works, is there a reason why creating a new polymorphic class would be better than modifying the current `\LORIS\api\Views\Visit` class?

Reply:
> you mean we could create a new Class \LORIS\api\Views\Visit that would take $version as a parameter?  
 
No, I mean that depending on `$version` value, the JsonResponse would take a `\LORIS\api\Views\Visit` or a `\LORIS\api\Views\Visit_0_0_2` object as a parameter. 
With PHP, we can't have 2 classes with the same name in the same namespace. 
There are many ways we could solve that. Factory pattern seems overkill here so just having  `switch` on the version should be enough.

> is there a reason why creating a new polymorphic class would be better than modifying the current \LORIS\api\Views\Visit class?

Modifying the existing Visit class adds to the number of things that class needs to handle.

Reply:
OK, I think I understand the advantage of doing it the way you are suggesting.

I thought it was unnecessary and redundant to create a whole new class that is almost identical except for a small difference. 

However, if I understand correctly, when the versions will start accumulating, adding parameters and exceptions to a single class to deal with all the different versions is likely to become more cumbersome and convoluted (in this case by adding the element `Project` when it's not v0.0.2).

Reply:
This is exacly what I am foreseeing.
Since this seems to be the first of probably many divergence in supported versions, maybe @driusan can add his 2 cents?